packages/components/src/components/form/ButtonIcon/ButtonIcon.tsx

```diff
@@ -14,7 +14,7 @@ export const ButtonIcon: FC<ButtonIconProps> = forwardRef<ButtonIconProps, 'butt
       isLoading: props.isLoading,
     }) as ButtonIconStyle;
 
-    const Icon = React.cloneElement(icon, { size: responsiveSize });
+    const Icon = React.cloneElement(icon, { size: responsiveSize, display: 'block' });
```

Comment:
📣 Setting display: block; on the <svg> changes it from inline to block, removing extra inline spacing and making the parent <span> fit the SVG's exact dimensions.